plugins/module_utils/mso.py

```diff
@@ -303,43 +305,50 @@ def __init__(self, module):
         self.response = None
         self.status = None
         self.url = None
+        self.nd_base_path = None
         self.httpapi_logs = list()
 
         if self.module._debug:
             self.module.warn("Enable debug output because ANSIBLE_DEBUG was set.")
             self.params["output_level"] = "debug"
 
+        # Ensure protocol is set
+        self.params["protocol"] = "https" if self.params.get("use_ssl", True) else "http"
+
+        if self.params.get("host") is None:
+            self.fail_json(msg="Parameter 'host' is required when not using the HTTP API connection plugin")
+
+        # Set base_uri
+        if self.params.get("port") is not None:
+            self.base_only_uri = "{protocol}://{host}:{port}/".format(**self.params)
+        else:
+            self.base_only_uri = "{protocol}://{host}/".format(**self.params)
+
         if self.module._socket_path is None:
+            # Set MSO params
             if self.params.get("use_ssl") is None:
                 self.params["use_ssl"] = True
             if self.params.get("use_proxy") is None:
                 self.params["use_proxy"] = True
             if self.params.get("validate_certs") is None:
                 self.params["validate_certs"] = True
 
-            # Ensure protocol is set
-            self.params["protocol"] = "https" if self.params.get("use_ssl", True) else "http"
-
-            # Set base_uri
-            if self.params.get("port") is not None:
-                self.base_only_uri = "{protocol}://{host}:{port}/".format(**self.params)
-                self.baseuri = "{0}api/v1/".format(self.base_only_uri)
-            else:
-                self.base_only_uri = "{protocol}://{host}/".format(**self.params)
-                self.baseuri = "{0}api/v1/".format(self.base_only_uri)
-
-            if self.params.get("host") is None:
-                self.fail_json(msg="Parameter 'host' is required when not using the HTTP API connection plugin")
+            self.baseuri = "{0}api/v1/".format(self.base_only_uri)
 
             if self.params.get("password"):
                 # Perform password-based authentication, log on using password
                 self.login()
             else:
                 self.fail_json(msg="Parameter 'password' is required for authentication")
         else:
+            # Set NDO params
             self.connection = Connection(self.module._socket_path)
             if self.connection.get_platform() == "cisco.nd":
                 self.platform = "nd"
+                self.nd_base_path = "mso/api/v1/"
```

Comment:
I don't think we should define this here for ND. I think we could use a similar way as done in request().
The code in request allow for v2 endpoints which exists in NDO.